src/porepy/numerics/ad/operators.py

```diff
@@ -1431,6 +1465,30 @@ def shape(self) -> tuple[int, int]:
         """Shape of the wrapped matrix."""
         return self._shape
 
+    @staticmethod
+    def _compute_spmatrix_hash(mat: sps.spmatrix) -> str:
+        """Utility function that handles all the sparse formats to compute the hash."""
```

Comment:
As a service to the future us, perhaps give some reasoning behind the design of the hash for this class. It is less obvious than the other classes.

Reply:
Provided, please take a look